src/com/magento/idea/magento2plugin/actions/generation/dialog/NewSetupDataPatchDialog.java

```diff
@@ -0,0 +1,223 @@
+
+        // call onCancel() when cross is clicked
+        setDefaultCloseOperation(DO_NOTHING_ON_CLOSE);
+        addWindowListener(new WindowAdapter() {
+            @Override
+            public void windowClosing(final WindowEvent event) {
+                onCancel();
+            }
+        });
+
+        // call onCancel() on ESCAPE
+        contentPanel.registerKeyboardAction(
+                (final ActionEvent event) -> onCancel(),
+                KeyStroke.getKeyStroke(KeyEvent.VK_ESCAPE, 0),
+                JComponent.WHEN_ANCESTOR_OF_FOCUSED_COMPONENT
+        );
+
+        addComponentListener(
+                new FocusOnAFieldListener(() -> className.requestFocusInWindow())
+        );
+    }
+
+    /**
+     * Open dialog.
+     *
+     * @param project Project
+     * @param directory PsiDirectory
+     */
+    public static void open(
+            final Project project,
+            final PsiDirectory directory,
+            final String modulePackage,
+            final String moduleName
+    ) {
+        final NewSetupDataPatchDialog dialog = new NewSetupDataPatchDialog(
+                project,
+                directory,
+                modulePackage,
+                moduleName
+        );
+        dialog.pack();
+        dialog.centerDialog(dialog);
+        dialog.setVisible(true);
+    }
+
+    protected void onOK() {
+        if (validateFields()) {
+            generateFile();
+            exit();
+        }
+    }
+
+    private void generateFile() {
+        final ModuleSetupDataPatchGenerator generator = new ModuleSetupDataPatchGenerator(
+                new ModuleSetupDataPatchData(
+                        modulePackage,
+                        moduleName,
+                        createDirectory(baseDir),
+                        getClassName()
+                ),
+                project
+        );
+
+        generator.generate(NewSetupDataPatchAction.ACTION_NAME, true);
+    }
+
+    public String getClassName() {
+        return className.getText().trim();
+    }
+
+    private boolean validateFields() {
+        final PsiDirectory targetDirectory = getDirectory(baseDir);
+
+        if (NewSetupDataPatchAction.DATA_DIRECTORY.equals(targetDirectory.getName())) {
+            final PsiFile[] files = targetDirectory.getFiles();
+            for (final PsiFile file : files) {
+                final String className = ModuleSetupDataPatchFile.resolveClassNameFromInput(
+                        getClassName()
+                );
+
+                if (file.getName().equals(className + ".php")) {
+                    showErrorMessage(
+                            fieldsValidationsList.get(0).getField(),
+                            "Class name `" + className + "` already exist."
+                    );
+
+                    return false;
+                }
+            }
+        }
+        return validateFormFields();
+    }
+
+    private PsiDirectory getDirectory(final PsiDirectory targetDirectory) {
```

Comment:
Please, simplify this logic. Make sure that we really need that.

Reply:
Done.